proteuslib/unit_models/reverse_osmosis_1D.py

```diff
@@ -0,0 +1,800 @@
+                                     .format(len(self.solvent_list)))
+
+        if self.config.transformation_method is useDefault:
+            _log.warning(
+                "Discretization method was "
+                "not specified for the "
+                "reverse osmosis module. "
+                "Defaulting to finite "
+                "difference method."
+            )
+            self.config.transformation_method = "dae.finite_difference"
+
+        if self.config.transformation_scheme is useDefault:
+            _log.warning(
+                "Discretization scheme was "
+                "not specified for the "
+                "reverse osmosis module."
+                "Defaulting to backward finite "
+                "difference."
+            )
+            self.config.transformation_scheme = "BACKWARD"
+
+
+    def build(self):
+        """
+        Build 1D RO model (pre-DAE transformation).
+
+        Args:
+            None
+
+        Returns:
+            None
+        """
+        # Call UnitModel.build to setup dynamics
+        super().build()
+
+        self.scaling_factor = Suffix(direction=Suffix.EXPORT)
+
+        self.solvent_list = Set()
+        self.solute_list = Set()
+        self._process_config()
+
+        # Build 1D Control volume for feed side
+        self.feed_side = ControlVolume1DBlock(default={
+            "dynamic": self.config.dynamic,
+            "has_holdup": self.config.has_holdup,
+            "area_definition": self.config.area_definition,
+            "property_package": self.config.property_package,
+            "property_package_args": self.config.property_package_args,
+            "transformation_method": self.config.transformation_method,
+            "transformation_scheme": self.config.transformation_scheme,
+            "finite_elements": self.config.finite_elements,
+            "collocation_points": self.config.collocation_points
+        })
+
+        feed_side = self.feed_side
+        # Add geometry to feed side
+        feed_side.add_geometry()
+        # Add state blocks to feed side
+        feed_side.add_state_blocks(has_phase_equilibrium=False)
+        # Populate feed side
+        feed_side.add_material_balances(balance_type=self.config.material_balance_type,
+                                        has_mass_transfer=True)
+        feed_side.add_energy_balances(balance_type=EnergyBalanceType.none)
+        feed_side.add_momentum_balances(balance_type=self.config.momentum_balance_type,
+                                        has_pressure_change=self.config.has_pressure_change)
+        # Apply transformation to feed side
+        feed_side.apply_transformation()
+        # Add inlet/outlet ports for feed side
+        self.add_inlet_port(name="feed_inlet", block=feed_side)
+        self.add_outlet_port(name="feed_outlet", block=feed_side)
+        #TODO: Make permeate_side.permeate_out a ProductBlock instead?
```

Comment:
No, you don't want a ProductBlock (although it would work). Feed and Product blocks are just wrappers around a StateBlock anyway, with an appropriately named Port (inlet for a Product block) and some extra code to allow users to do `Product.state_var.fix`.